src/index.js

```diff
@@ -137,8 +137,16 @@ const unexpectedEnzyme = {
     childExpect.exportAssertion(
       '<ReactWrapper> [not] to have class <string>',
       (expect, reactWrapper, className) => {
-        if (expect.flags.not === reactWrapper.hasClass(className)) {
-          const classes = (reactWrapper.prop('className') || '').split(' ');
+        const getFirstNode = wrapper => {
+          return typeof wrapper.type() === 'function'
+            ? getFirstNode(wrapper.childAt(0))
+            : wrapper;
```

Comment:
@sunesimonsen this is kind of weird, but if you render something like https://github.com/albertfdp/unexpected-enzyme/pull/11/files#diff-bd93117e77914cdc01207a55e1736ca9R5 then enzyme says that `wrapper.hasClass('root')` is `false` ¯\_(ツ)_/¯

Reply:
woot :-S

Reply:
Yeah, because it considers it the root element:

```jsx
<Fixture>
 <div className="root top">
  ...
 </div>
</Fixture>
```

I think it's fair to say that you are usually trying to assert on the `<div/>` but we might be shooting ourselves on the foot here. It might be better to just  do what `enzyme` does...

Reply:
It might be a bad idea to find the first non-composed component, because we are changing the normal behaviour of enzyme.

Reply:
I agree.

Reply:
Actually it works in `chai-enzyme` https://github.com/producthunt/chai-enzyme/blob/master/test/className.test.js#L16 but they are using `.classNames()` https://github.com/producthunt/chai-enzyme/blob/master/src/assertions/className.js#L2 which does not exist in enzyme's API, will see how do they get it.

Reply:
Much better now https://github.com/albertfdp/unexpected-enzyme/pull/11/commits/534a1e7c59d74246b0f39d33e36468238b853e86